src/main/java/seedu/algobase/logic/commands/MoveTaskCommand.java

```diff
@@ -0,0 +1,123 @@
+        if (moveTaskDescriptor.planFromIndex.getZeroBased() >= lastShownPlanList.size()
+            || moveTaskDescriptor.planToIndex.getZeroBased() >= lastShownPlanList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_TASK_DISPLAYED_INDEX);
+        }
+
+        Plan planFrom = lastShownPlanList.get(moveTaskDescriptor.planFromIndex.getZeroBased());
+        List<Task> taskListFrom = new ArrayList<>(planFrom.getTasks());
+        Task taskToMove = taskListFrom.get(moveTaskDescriptor.taskIndex.getZeroBased());
+        taskListFrom.remove(moveTaskDescriptor.taskIndex.getZeroBased());
+        Set<Task> taskSetFrom = new HashSet<>(taskListFrom);
+        Plan updatedPlanFrom = Plan.updateTasks(planFrom, taskSetFrom);
+
+        Plan planTo = lastShownPlanList.get(moveTaskDescriptor.planToIndex.getZeroBased());
+        List<Task> taskListTo = new ArrayList<>(planTo.getTasks());
+        Set<Task> taskSetTo = new HashSet<>(taskListTo);
+        if (taskSetTo.contains(taskToMove)) {
+            return new CommandResult(
+                String.format(MESSAGE_DUPLICATE_TASK, taskToMove.getProblem().getName(), planTo.getPlanName()));
+        }
+        taskSetTo.add(taskToMove);
+        Plan updatedPlanTo = Plan.updateTasks(planTo, taskSetTo);
+
+        model.setPlan(planFrom, updatedPlanFrom);
+        model.setPlan(planTo, updatedPlanTo);
+        model.updateFilteredPlanList(PREDICATE_SHOW_ALL_PLANS);
+        return new CommandResult(
+            String.format(MESSAGE_MOVE_TASK_SUCCESS,
+                taskToMove.getProblem().getName(),
+                updatedPlanFrom.getPlanName(),
+                updatedPlanTo.getPlanName()
+            )
+        );
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof MoveTaskCommand // instanceof handles nulls
+                && moveTaskDescriptor.equals(((MoveTaskCommand) other).moveTaskDescriptor)); // state check
+    }
+
+    /**
+     * Stores the details of the plan and problem involved.
+     */
+    public static class MoveTaskDescriptor {
+        private Index taskIndex;
+        private Index planFromIndex;
+        private Index planToIndex;
+
+        public MoveTaskDescriptor(Index taskIndex, Index planFromIndex, Index planToIndex) {
+            this.taskIndex = taskIndex;
+            this.planFromIndex = planFromIndex;
+            this.planToIndex = planToIndex;
+        }
+
+        @Override
+        public boolean equals(Object other) {
+            return other == this // short circuit if same object
+                || (other instanceof MoveTaskDescriptor // instanceof handles nulls
+                && taskIndex.equals(((MoveTaskDescriptor) other).taskIndex)
+                && planFromIndex.equals(((MoveTaskDescriptor) other).planFromIndex)
+                && planToIndex.equals(((MoveTaskDescriptor) other).planToIndex));
+        }
+    }
```

Comment:
Not sure if it's justifiable to use `Descriptor` in this case, considering how you implement `MoveTaskCommandParser`, is it also possible to move the content inside `MoveTaskDescriptor` to the parameters in the constructor of `MoveTaskCommand`?